### PR TITLE
VCITA-39767 - Voice Call Setting Entity

### PR DESCRIPTION
**Pull request by MahmudQasemVcita**

### CODE REVIEW

File: entities/communication/voiceCallSetting.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            "type": "string",
+            "description": "Staff availability uid."
+        },
+        "forward_call": {
+            "type": "enum",
+            "values": [
+                "ALWAYS",
+                "WORK_HOURS",
+                "NEVER"],
+            "description": "Forward call option based on staff availability."
```
Comment (eliran-vcita) [inline entities/communication/voiceCallSetting.json:30]:
```suggestion
        "call_forwarding_schedule": {
            "type": "enum",
            "values": [
                "ALWAYS",
                "WORK_HOURS",
                "NEVER"],
            "description": "When to forward the call based on staff availability."
```

Comment (MahmudQasemVcita) [reply entities/communication/voiceCallSetting.json:30]: @eliran-vcita , Can be staff_forwarding_availability more accurate? as we are storing availability of the staff, 24/7 - based on working-hours -  not available to receive calls.

Comment (eliran-vcita) [reply entities/communication/voiceCallSetting.json]: Is this the flow?
call comes in => check `call_forwarding_schedule`
if `ALWAYS` => forward call to staff (no matter what their availability is)
if `WORK_HOURS` => forward call based on availability
if `NEVER` => don't forward?

Also - note to @alonyacobi - let's settle on a convention for enum values. I like this one with uppercase and snakecase values.

Comment (MahmudQasemVcita) [reply entities/communication/voiceCallSetting.json]: Indeed @eliran-vcita , that's why i an not sure about schedule, maybe even call_forwarding_preference?

Comment (eliran-vcita) [reply entities/communication/voiceCallSetting.json]: preference is another word for setting 😄 

if this is the flow - I recommend `call_forwarding_policy`

File: entities/communication/voiceCallSetting.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
+{
+    "type": "object",
+    "properties": {
+        "uid": {
+            "type": "string",
+            "description": "The voice call settings unique identifier."
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:6]: It reads strange, please make it "The VoiceCallSetting unique identifier (its also singular not plural)

File: entities/communication/voiceCallSetting.json [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        "staff_uid": {
+            "type": "string",
+            "description": "Receiving forwarded call staff uid."
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:10]: The unique identifier of the staff receiving the forwarded call

File: entities/communication/voiceCallSetting.json [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        "forward_number": {
+            "type": "string",
+            "description": "The phone number of the staff receiving the forwarded call."
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:14]: remove all periods at the end of descriptions

File: entities/communication/voiceCallSetting.json [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        "dedicated_number": {
+            "type": "string",
+            "description": "The dedicated number of the business receiving calls from clients."
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:18]: remove "from clients" (you don't know its clients calling)

File: entities/communication/voiceCallSetting.json [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        "forwarding_enabled": {
+            "type": "boolean",
+            "description": "Is forwarding the voice call enabled or not."
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:22]: Indicates if voice call forwarding is enabled

File: entities/communication/voiceCallSetting.json [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        "call_forwarding_policy": {
+            "type": "enum",
+            "values": [
+                "ALWAYS",
+                "WORK_HOURS",
+                "NEVER"],
+            "description": "When to forward the call based on staff availability."
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:30]: Specifies the conditions under which calls are forwarded based on staff availability

File: entities/communication/voiceCallSetting.json [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        "staff_weekly_availability_uid": {
+            "type": "string",
+            "description": "Staff weekly availability details uid."
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:34]: I don't understand what this is doing? is this a reference to a value object. What do we have in that value object. How does it impact functionality?

File: entities/communication/voiceCallSetting.json [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        "call_timeout_sec": {
+            "type": "number",
+            "description": "Call ringing timeout in seconds."
+        },
+        "voice_scripts": {
+            "type": "json",
+            "description": "Voice scripts to play for caller."
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:42]: A key-value JSON object defining the voice scripts to play for the caller. 

What are the available keys here? is it open, how does it work?

File: entities/communication/voiceCallSetting.json [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        "external_app_config": {
+            "type": "json",
+            "description": "External AI app details."
+        },
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:47]: A configuration JSON object defining the external AI app managing the voice call. The object defines three keys:
 app_name: the name of the app used (for internal use)
 full_url: the endpoint related to that app
 method: the HTTP method that should be used (GET/POST)

File: entities/communication/voiceCallSetting.json [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+        "created_at": {
+            "type": "timestamp",
+            "description": "The creation date and time of the voice call settings.",
+            "format": "date-time"
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:51]: The date and time when the VoiceCallSetting entity was created

File: entities/communication/voiceCallSetting.json [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        "updated_at": {
+            "type": "timestamp",
+            "description": "Updated date and time of the voice call settings.",
+            "format": "date-time"
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:56]: The date and time when the VoiceCallSetting entity was updated

File: entities/communication/voiceCallSetting.json [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+        },
+        "deleted_at": {
+            "type": "timestamp",
+            "description": "The deletion date and time of the voice call settings.",
+            "format": "date-time"
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:61]: The date and time when the VoiceCallSetting entity was deleted

**Review by ramsterita — CHANGES_REQUESTED** [review]

File: entities/communication/voiceCallSetting.json [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+            "type": "string",
+            "description": "The unique identifier represents the weekly working hours availability list object of staff members\nThis list will assist deciding if to forward the call to staff or its out of the working hours (in case forwarding policy is WORK_HOURS)."
+        },
+        "call_timeout_sec": {
+            "type": "number",
+            "description": "Call ringing timeout in seconds"
+        },
+        "voice_scripts": {
+            "type": "json",
+            "description": "A key-value JSON object defining the voice scripts to play for the caller when text to speech action required, for example: greeting message\nKey values of the voice scripts most be one of:\n- greeting_message\n- missing_call_message"
```
Comment (ramsterita) [inline entities/communication/voiceCallSetting.json:42]: must be (not most be)

**Review by ramsterita — APPROVED** [review]

**Pull request merged**